chia/data_layer/download_data.py

```diff
@@ -0,0 +1,184 @@
+from chia.data_layer.data_store import DataStore
+from chia.types.blockchain_format.sized_bytes import bytes32
+
+
+def get_full_tree_filename(tree_id: bytes32, node_hash: bytes32, generation: int) -> str:
+    return f"{tree_id}-{node_hash}-full-{generation}-v1.0.dat"
+
+
+def get_delta_filename(tree_id: bytes32, node_hash: bytes32, generation: int) -> str:
+    return f"{tree_id}-{node_hash}-delta-{generation}-v1.0.dat"
+
+
+def is_filename_valid(filename: str) -> bool:
+    split = filename.split("-")
+
+    try:
+        raw_tree_id, raw_node_hash, file_type, raw_generation, raw_version, *rest = split
+        tree_id = bytes32(bytes.fromhex(raw_tree_id))
+        node_hash = bytes32(bytes.fromhex(raw_node_hash))
+        generation = int(raw_generation)
+    except ValueError:
+        return False
+
+    if len(rest) > 0:
+        return False
+
+    # TODO: versions should probably be centrally defined
+    if raw_version != "v1.0.dat":
+        return False
+
+    if file_type not in {"delta", "full"}:
+        return False
+
+    generate_file_func = get_delta_filename if file_type == "delta" else get_full_tree_filename
+    reformatted = generate_file_func(tree_id=tree_id, node_hash=node_hash, generation=generation)
+
+    return reformatted == filename
+
+
+async def insert_into_data_store_from_file(
+    data_store: DataStore,
+    tree_id: bytes32,
+    root_hash: Optional[bytes32],
+    filename: Path,
+) -> None:
+    with open(filename, "rb") as reader:
+        while True:
+            chunk = b""
+            while len(chunk) < 4:
+                size_to_read = 4 - len(chunk)
+                cur_chunk = reader.read(size_to_read)
+                if cur_chunk is None or cur_chunk == b"":
+                    if size_to_read < 4:
+                        raise Exception("Incomplete read of length.")
+                    break
+                chunk += cur_chunk
+            if chunk == b"":
+                break
+
+            size = int.from_bytes(chunk, byteorder="big")
+            serialize_nodes_bytes = b""
+            while len(serialize_nodes_bytes) < size:
+                size_to_read = size - len(serialize_nodes_bytes)
+                cur_chunk = reader.read(size_to_read)
+                if cur_chunk is None or cur_chunk == b"":
+                    raise Exception("Incomplete read of blob.")
+                serialize_nodes_bytes += cur_chunk
+            serialized_node = SerializedNode.from_bytes(serialize_nodes_bytes)
+
+            node_type = NodeType.TERMINAL if serialized_node.is_terminal else NodeType.INTERNAL
+            await data_store.insert_node(node_type, serialized_node.value1, serialized_node.value2)
+
+    await data_store.insert_root_with_ancestor_table(tree_id=tree_id, node_hash=root_hash, status=Status.COMMITTED)
+
+
+async def write_files_for_root(
+    data_store: DataStore,
+    tree_id: bytes32,
+    root: Root,
+    foldername: Path,
+    override: bool = False,
```

Comment:
```suggestion
    overwrite: bool = False,
```

`override = True` is a lot bigger danger to their data; better to make this flag stand out.

Reply:
Made a PR to change `override` to `overwrite`: https://github.com/Chia-Network/chia-blockchain/pull/13224